File: speech-to-text/getting-started/python/transcribe.py
#!/usr/bin/env python3
"""
Smallest AI Speech-to-Text - Getting Started

The simplest way to transcribe audio using Smallest AI's Pulse STT API.

Usage: python transcribe.py <audio_file>

Output:
- Command line response with transcription
"""

import os
import sys
import requests

API_URL = "https://waves-api.smallest.ai/api/v1/pulse/get_text"

# Features
LANGUAGE = "en"  # Use ISO 639-1 codes or "multi" for auto-detect


def transcribe(audio_file: str, api_key: str) -> dict:
    with open(audio_file, "rb") as f:
        audio_data = f.read()

    response = requests.post(
        API_URL,
        headers={
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/octet-stream",
        },
        params={
            "language": LANGUAGE,
        },
        data=audio_data,
    )

    if response.status_code != 200:
        raise Exception(f"API request failed with status {response.status_code}: {response.text}")

    return response.json()


def process_response(result: dict):
    if result.get("status") != "success":
        print(f"Error: Transcription failed")
        print(result)
        sys.exit(1)

    print("\n" + "=" * 60)
    print("TRANSCRIPTION")
    print("=" * 60)
    print(result.get("transcription", ""))
    print("=" * 60 + "\n")


def main():
    if len(sys.argv) < 2:
        print("Usage: python transcribe.py <audio_file>")
        sys.exit(1)

    audio_file = sys.argv[1]
    api_key = os.environ.get("SMALLEST_API_KEY")

    if not api_key:
        print("Error: SMALLEST_API_KEY environment variable not set")
        sys.exit(1)

    if not os.path.exists(audio_file):
        print(f"Error: File not found: {audio_file}")
        sys.exit(1)

    result = transcribe(audio_file, api_key)
    process_response(result)


if __name__ == "__main__":
    main()

File: speech-to-text/getting-started/javascript/transcribe.js
#!/usr/bin/env node
/**
 * Smallest AI Speech-to-Text - Getting Started
 *
 * The simplest way to transcribe audio using Smallest AI's Pulse STT API.
 *
 * Usage: node transcribe.js <audio_file>
 *
 * Output:
 * - Command line response with transcription
 */

const fs = require("fs");

const API_URL = "https://waves-api.smallest.ai/api/v1/pulse/get_text";

// Features
const LANGUAGE = "en"; // Use ISO 639-1 codes or "multi" for auto-detect

async function transcribe(audioFile, apiKey) {
  const audioData = fs.readFileSync(audioFile);

  const params = new URLSearchParams({
    language: LANGUAGE,
  });

  const response = await fetch(`${API_URL}?${params}`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${apiKey}`,
      "Content-Type": "application/octet-stream",
    },
    body: audioData,
  });

  if (!response.ok) {
    throw new Error(`API request failed with status ${response.status}: ${await response.text()}`);
  }

  return response.json();
}

function processResponse(result) {
  if (result.status !== "success") {
    console.error("Error: Transcription failed");
    console.error(result);
    process.exit(1);
  }

  console.log("");
  console.log("=".repeat(60));
  console.log("TRANSCRIPTION");
  console.log("=".repeat(60));
  console.log(result.transcription || "");
  console.log("=".repeat(60));
  console.log("");
}

async function main() {
  const audioFile = process.argv[2];

  if (!audioFile) {
    console.log("Usage: node transcribe.js <audio_file>");
    process.exit(1);
  }

  const apiKey = process.env.SMALLEST_API_KEY;

  if (!apiKey) {
    console.error("Error: SMALLEST_API_KEY environment variable not set");
    process.exit(1);
  }

  if (!fs.existsSync(audioFile)) {
    console.error(`Error: File not found: ${audioFile}`);
    process.exit(1);
  }

  const result = await transcribe(audioFile, apiKey);
  processResponse(result);
}

main();
